junifer/api/queue_context/gnu_parallel_local_adapter.py

```diff
@@ -0,0 +1,171 @@
+from .queue_context_adapter import QueueContextAdapter
+
+
+__all__ = ["GnuParallelLocalAdapter"]
+
+
+class GnuParallelLocalAdapter(QueueContextAdapter):
+    """Class for generating commands for GNU Parallel (local).
+
+    Parameters
+    ----------
+    job_name : str
+        The job name.
+    job_dir : pathlib.Path
+        The path to the job directory.
+    yaml_config_path : pathlib.Path
+        The path to the YAML config file.
+    elements : list of str or tuple
+        Element(s) to process. Will be used to index the DataGrabber.
+    env : dict, optional
+        The Python environment configuration. If None, will run without a
+        virtual environment of any kind (default None).
+    verbose : str, optional
+        The level of verbosity (default "info").
+    submit : bool, optional
+        Whether to submit the jobs (default False).
+
+    Raises
+    ------
+    ValueError
+        If``env`` is invalid.
+
+    See Also
+    --------
+    QueueContextAdapter :
+        The base class for QueueContext.
+    HTCondorAdapter :
+        The concrete class for queueing via HTCondor.
+
+    """
+
+    def __init__(
+        self,
+        job_name: str,
+        job_dir: Path,
+        yaml_config_path: Path,
+        elements: List[Union[str, Tuple]],
+        env: Optional[Dict[str, str]] = None,
+        verbose: str = "info",
+        submit: bool = False,
+    ) -> None:
+        """Initialize the class."""
+        self._job_dir = job_dir
+        self._yaml_config_path = yaml_config_path
+        self._elements = elements
+        self._check_env(env)
+        self._verbose = verbose
+        self._submit = submit
+
+        self._run_joblog_path = self._job_dir / f"{job_name}_run_joblog.txt"
+
+    def _check_env(self, env: Optional[Dict[str, str]]) -> None:
+        """Check value of env parameter on init.
+
+        Parameters
+        ----------
+        env : dict or None
+            The value of env parameter.
+
+        Raises
+        ------
+        ValueError
+            If ``env.kind`` is invalid.
+
+        """
+        # Set env related variables
+        if env is None:
+            env = {"kind": "local"}
+        # Check env kind
+        valid_env_kinds = ["conda", "venv", "local"]
+        if env["kind"] not in valid_env_kinds:
+            raise_error(
+                f"Invalid value for `env.kind`: {env['kind']}, "
+                f"must be one of {valid_env_kinds}"
+            )
+        else:
+            # Set variables
+            if env["kind"] == "local":
+                # No virtual environment
+                self._executable = "junifer"
+                self._arguments = ""
+            else:
+                self._executable = f"run_{env['kind']}.sh"
+                self._arguments = f"{env['name']} junifer"
+                self._exec_path = self._job_dir / self._executable
+
+    def run(self) -> str:
+        """Return run commands."""
+        junifer_run_cmd = (
```

Comment:
Some comments about the `parallel` arguments.
`--bar`: ok, shows a bar.
`--halt`: disagree, we should run as much as we can. If one subject fails, continue.
`--resume`: add this so the user sees the full command that can be stopped and continued (with ctrl+C)
`--resume-failed`: add this also, for the same reason. It will not have any effect on the first run. It will have an effect on the subsequent runs.
`--joblob`: perfect


Some other to consider:
`--output-as-files`: to redirect output to a file and keep the stdout clean
`--delay`: prevent having N-jobs doing the same IO operation at the beginning, which will definitely create a bottleneck and maybe a failure.

Reply:
Agreed. Any reason for preferring `--output-as-files` over `--results`? The latter stores the stdout, stderr and seq value in a structured way.

Reply:
Did not know of the existence of `--results`. Pick what you think it's the best option.